src/filerepository.js

```diff
@@ -354,6 +373,25 @@ class FileLoader {
 		this.set( 'uploadResponse', null );
 	}
 
+	/**
+	 * Returns a `Promise` which resolves to a `File` instance associated with this file loader.
+	 *
+	 * @type {Promise}
+	 */
+	get file() {
+		if ( !this._filePromiseWrapper ) {
+			// Loader was destroyed, return promise which resolves to null.
+			return new Promise( resolve => resolve( null ) );
+		} else {
+			// The `loader.file.then( ... )` can be called, then `loader._destroy()` (which is sync) and then
+			// `loader.file.then( ... )` will be resolved. In such case, even after `destroy` call the promise will
+			// return `File` instance so we use nested promises which handles this case.
+			return this._filePromiseWrapper.promise.then( file => {
+				return new Promise( resolve => resolve( this._filePromiseWrapper ? file : null ) );
```

Comment:
Similarly to `if()` this can be shortened:
```js
return this._filePromiseWrapper.promise
    .then( file => this._filePromiseWrapper ? file : null );
```
as the promise from `this._filePromiseWrapper` will be used to create a promise chain and the returned value will passed to the `.then()` call on `loader.file.then()`.